src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -35,17 +43,63 @@
     WrappedGenerateLockfile,
 )
 from pants.core.util_rules.lockfile_metadata import calculate_invalidation_digest
-from pants.engine.fs import CreateDigest, Digest, DigestContents, FileContent, MergeDigests
+from pants.engine.fs import (
+    CreateDigest,
+    Digest,
+    DigestContents,
+    DigestEntries,
+    FileContent,
+    FileEntry,
+    GlobMatchErrorBehavior,
+    MergeDigests,
+    PathGlobs,
+)
 from pants.engine.internals.synthetic_targets import SyntheticAddressMaps, SyntheticTargetsRequest
 from pants.engine.internals.target_adaptor import TargetAdaptor
 from pants.engine.process import ProcessCacheScope, ProcessResult
 from pants.engine.rules import Get, collect_rules, rule
 from pants.engine.target import AllTargets
 from pants.engine.unions import UnionRule
+from pants.option.option_types import BoolOption, StrListOption
+from pants.option.subsystem import Subsystem
 from pants.util.docutil import bin_name
 from pants.util.logging import LogLevel
 from pants.util.ordered_set import FrozenOrderedSet
 from pants.util.pip_requirement import PipRequirement
+from pants.util.strutil import softwrap
+
+logger = logging.getLogger(__name__)
+
+
+class PexLockSubsystem(Subsystem):
+    options_scope = "pex-lock"
+    help = "Pex lock specific Python lockfile options."
+
+    update = BoolOption(
+        default=False,
+        advanced=False,
+        help=softwrap(
+            """
+            update instead of creating a new lockfile
+            """
+        ),
+    )
+
+    project = StrListOption(
```

Comment:
I imagine this would usually be passed on the command line, rather than set in `pants.toml`, e.g. `pants generate-lockfiles --resolve=... --pex-lock-project=some-library`. Is that how you'd see it being used?

Reply:
Yep.  I think cli invocation would be most common.